File: sdk/java/src/test/java/com/flagent/client/model/PutSegmentRequestTest.java
package com.flagent.client.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for PutSegmentRequest
 */
class PutSegmentRequestTest {

    @Test
    void testPutSegmentRequest() {
        PutSegmentRequest model = new PutSegmentRequest()
            .description("Updated segment")
            .rolloutPercent(75L);
        Assertions.assertNotNull(model);
        Assertions.assertEquals("Updated segment", model.getDescription());
        Assertions.assertEquals(75L, model.getRolloutPercent());
    }

    @Test
    void descriptionTest() {
        PutSegmentRequest model = new PutSegmentRequest().description("Desc");
        Assertions.assertEquals("Desc", model.getDescription());
    }

    @Test
    void rolloutPercentTest() {
        PutSegmentRequest model = new PutSegmentRequest().rolloutPercent(50L);
        Assertions.assertEquals(50L, model.getRolloutPercent());
    }

}

File: sdk/dart/test/put_segment_request_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for PutSegmentRequest
void main() {
  final instance = PutSegmentRequest((b) => b
    ..description = 'Updated segment'
    ..rolloutPercent = 75);

  group(PutSegmentRequest, () {
    test('to test the property `description`', () async {
      expect(instance.description, equals('Updated segment'));
    });

    test('to test the property `rolloutPercent`', () async {
      expect(instance.rolloutPercent, equals(75));
    });
  });
}
